hydra-kafka10/src/main/java/com/addthis/hydra/kafka/bundle/KafkaSource.java

```diff
@@ -0,0 +1,283 @@
+
+import java.io.File;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.Properties;
+
+import com.addthis.basis.util.LessFiles;
+
+import com.addthis.bark.ZkUtil;
+import com.addthis.bundle.channel.DataChannelError;
+import com.addthis.bundle.core.Bundle;
+import com.addthis.bundle.core.list.ListBundleFormat;
+import com.addthis.hydra.kafka.KafkaUtils;
+import com.addthis.hydra.store.db.DBKey;
+import com.addthis.hydra.store.db.PageDB;
+import com.addthis.hydra.task.run.TaskRunConfig;
+import com.addthis.hydra.task.source.SimpleMark;
+import com.addthis.hydra.task.source.TaskDataSource;
+import com.addthis.hydra.task.source.bundleizer.BundleizerFactory;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.curator.framework.CuratorFramework;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.addthis.hydra.kafka.bundle.BundleWrapper.bundleQueueEndMarker;
+
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class KafkaSource extends TaskDataSource {
+
+    private static final Logger log = LoggerFactory.getLogger(KafkaSource.class);
+
+    @JsonProperty(required = true)
+    private String zookeeper;
+    @JsonProperty(required = true)
+    private String topic;
+    @JsonProperty
+    private String initialOffset = "earliest";
+    @JsonProperty
+    private String dateFormat = "YYMMdd";
+    @JsonProperty
+    private String markDir = "marks";
+    /** Ignore the mark directory */
+    @JsonProperty
+    private boolean ignoreMarkDir;
+    /** Specifies conversion to bundles.  If null, then uses DataChannelCodec.decodeBundle */
+    @JsonProperty
+    private BundleizerFactory format;
+
+    @JsonProperty
+    private int fetchThreads = 1;
+    @JsonProperty
+    private int decodeThreads = 1;
+    @JsonProperty
+    private int queueSize = 10000;
+    @JsonProperty
+    private int seedBrokers = 3;
+
+    @JsonProperty
+    private TaskRunConfig config;
+    @JsonProperty
+    private int pollRetries = 180;
+
+    @JsonProperty
+    private String bootstrapServers;
+    @JsonProperty
+    private Map<String, String> overrideProperties = new HashMap();
```

Comment:
I would name this field just "overrides" to be more consistent with our other configs